### PR TITLE
Add the scale field to Mesh and Collada nodes

### PR DESCRIPTION
**Pull request by BenjaminHug8**
This PR solves #152 and #153.

Linked to https://github.com/cyberbotics/webots/pull/4243.

### CODE REVIEW

File: urdf2webots/writeRobot.py [hunk 1]
```diff
@@ -239,15 +239,18 @@ def URDFBoundingObject(robotFile, link, level, boxCollision):
...
             robotFile.write((boundingLevel + 1) * indent + 'rotation %lf %lf %lf %lf\n' % (boundingObject.rotation[0],
-                                                                                       boundingObject.rotation[1],
-                                                                                       boundingObject.rotation[2],
-                                                                                       boundingObject.rotation[3]))
+                                                                                        boundingObject.rotation[1],
+                                                                                        boundingObject.rotation[2],
+                                                                                        boundingObject.rotation[3]))
+            robotFile.write((boundingLevel + 1) * indent + 'scale %lf %lf %lf\n' % (boundingObject.scale[0],
+                                                                                    boundingObject.scale[1],
+                                                                                    boundingObject.scale[2]))
```
Comment (omichel) [inline urdf2webots/writeRobot.py:253]: I have a general remark here: if the value to be written for a translation, rotation or scale (or any other field) is the default value, then we shouldn't write it. That would make the proto output much cleaner and more readable.

**Review by omichel — APPROVED** [review]
That seems good to me.
I know my suggestion is not the purpose of this PR, but we could make it here if easy to do or in a separate PR.

File: urdf2webots/writeRobot.py (hunk 1 above)
Comment (BenjaminHug8) [reply urdf2webots/writeRobot.py:253]: Totally agree with that. I will do it here.

File: urdf2webots/writeRobot.py [hunk 2]
```diff
@@ -260,10 +268,10 @@ def URDFBoundingObject(robotFile, link, level, boxCollision):
                                                                                 boundingObject.geometry.box.z))
             robotFile.write(boundingLevel * indent + '}\n')
 
-        elif boundingObject.geometry.cylinder.radius != 0 and boundingObject.geometry.cylinder.length != 0:
+        elif boundingObject.geometry.cylinder.radius != 0 and boundingObject.geometry.cylinder.height != 0:
             robotFile.write(initialIndent + 'Cylinder {\n')
             robotFile.write((boundingLevel + 1) * indent + 'radius ' + str(boundingObject.geometry.cylinder.radius) + '\n')
-            robotFile.write((boundingLevel + 1) * indent + 'height ' + str(boundingObject.geometry.cylinder.length) + '\n')
+            robotFile.write((boundingLevel + 1) * indent + 'height ' + str(boundingObject.geometry.cylinder.height) + '\n')
```
Comment (omichel) [inline urdf2webots/writeRobot.py:274]: Also here we shouldn't write the radius if it is the default value. Same for the height.

File: urdf2webots/writeRobot.py [hunk 3]
```diff
@@ -374,7 +385,7 @@ def URDFVisual(robotFile, visualNode, level, normal=False):
         elif visualNode.geometry.cylinder.radius != 0:
             robotFile.write((shapeLevel + 1) * indent + 'geometry Cylinder {\n')
             robotFile.write((shapeLevel + 2) * indent + 'radius ' + str(visualNode.geometry.cylinder.radius) + '\n')
-            robotFile.write((shapeLevel + 2) * indent + 'height ' + str(visualNode.geometry.cylinder.length) + '\n')
+            robotFile.write((shapeLevel + 2) * indent + 'height ' + str(visualNode.geometry.cylinder.height) + '\n')
```
Comment (omichel) [inline urdf2webots/writeRobot.py:388]: Same here.

**Review by omichel — CHANGES_REQUESTED** [review]
My comments may also apply to other default values, like Sphere.radius, Box.size, etc.

**Review by omichel — APPROVED** [review]
Great. Thank you.

**Pull request merged**